scrapy_poet/injection.py

```diff
@@ -139,7 +139,11 @@ def build_plan(self, request: Request) -> andi.Plan:
             callback,
             is_injectable=is_injectable,
             externally_provided=self.is_class_provided_by_any_provider,
-            overrides=self.overrides_registry.overrides_for(request).get,
+            # Ignore the type since andi.plan expects overrides to be
+            # Callable[[Callable], Optional[Callable]] but the registry
+            # returns a more accurate typing for this scenario:
+            # Mapping[Type[ItemPage], Type[ItemPage]]
```

Comment:
Is it a typing issue to be fixed in andi?

Reply:
Yes, this part: [https://github.com/scrapinghub/andi/blob/master/andi/andi.py#L131](https://github.com/scrapinghub/andi/blob/master/andi/andi.py#L131)

Specifically, I think andi needs to support the signature of `dict.get()` as well.